spec/controllers/admin/bulk_line_items_controller_spec.rb

```diff
@@ -0,0 +1,256 @@
+        end
+
+        it "retrieves a list of line_items" do
+          keys = json_response.first.keys.map{ |key| key.to_sym }
+          line_item_attributes.all?{ |attr| keys.include? attr }.should == true
+        end
+      end
+
+      context "hub enterprise" do
+        before do
+          controller.stub spree_current_user: distributor1.owner
+          spree_get :index, :format => :json
+        end
+
+        it "retrieves a list of line_items" do
+          keys = json_response.first.keys.map{ |key| key.to_sym }
+          line_item_attributes.all?{ |attr| keys.include? attr }.should == true
+        end
+      end
+    end
+  end
+
+  describe '#update' do
+    let(:supplier) { create(:supplier_enterprise) }
+    let(:distributor1) { create(:distributor_enterprise) }
+    let(:coordinator) { create(:distributor_enterprise) }
+    let(:order_cycle) { create(:simple_order_cycle, coordinator: coordinator) }
+    let!(:order1) { FactoryGirl.create(:order, order_cycle: order_cycle, state: 'complete', completed_at: Time.zone.now, distributor: distributor1, billing_address: FactoryGirl.create(:address) ) }
+    let!(:line_item1) { FactoryGirl.create(:line_item, order: order1, product: FactoryGirl.create(:product, supplier: supplier)) }
+    let(:line_item_params) { { quantity: 3, final_weight_volume: 3000, price: 3.00 } }
+    let(:params) { { id: line_item1.id, order_id: order1.number, line_item: line_item_params } }
+
+    context "as an enterprise user" do
+      context "producer enterprise" do
+        before do
+          controller.stub spree_current_user: supplier.owner
+          spree_put :update, params
+        end
+
+        it "does not allow access" do
+          expect(response).to redirect_to spree.unauthorized_path
+        end
+      end
+
+      context "coordinator enterprise" do
+        render_views
+
+        before do
+          controller.stub spree_current_user: coordinator.owner
+        end
+
+        # Used in admin/orders/bulk_management
+        context 'when the request is JSON (angular)' do
+          before { params[:format] = :json }
+
+          it "updates the line item" do
+            spree_put :update, params
+            line_item1.reload
+            expect(line_item1.quantity).to eq 3
+            expect(line_item1.final_weight_volume).to eq 3000
+            expect(line_item1.price).to eq 3.00
+          end
+
+          it "returns an empty JSON response" do
+            spree_put :update, params
+            expect(response.body).to eq ' '
+          end
+
+          it 'returns a 204 response' do
+            spree_put :update, params
+            expect(response.status).to eq 204
+          end
+
+          it 'applies enterprise fees locking the order with an exclusive row lock' do
+            expect_any_instance_of(Spree::Order).to receive(:reload).with(lock: true)
+            expect_any_instance_of(Spree::Order).to receive(:update_distribution_charge!)
```

Comment:
Opened an issue to fix this https://github.com/openfoodfoundation/openfoodnetwork/issues/1762